Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp

```diff
@@ -582,10 +582,10 @@ int CLuaEngineDefs::EngineRequestModel(lua_State* luaVM)
             {
                 int iModelID = m_pManager->GetModelManager()->GetFirstFreeModelID();
                 if (iModelID != INVALID_MODEL_ID) {
-                    CClientModel* pModel = m_pManager->GetModelManager()->FindModelByID(iModelID);
+                    std::shared_ptr<CClientModel> pModel = m_pManager->GetModelManager()->FindModelByID(iModelID);
                     if (pModel == nullptr)
-                        pModel = new CClientModel(m_pManager, iModelID, eModelType);
-
+                        pModel = std::make_shared<CClientModel>(m_pManager, iModelID, eModelType);
+                    m_pManager->GetModelManager()->Add(pModel);
```

Comment:
Let's move this to `CClientModelManager` instead, this doesnt belong to the API code.